woocommerce/woocommerce-gutenberg-products-block
# Add Slot/Fill to discounts area in cart sidebar

This PR adds a Slot/Fill to the Cart sidebar (The same functionality is required in Checkout, but this will come in another PR). Extensions can register components to be fill this Slot, it is rendered just underneath the `TotalsCoupon`  component.

The PR also includes a (partial) revert of https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4238 (specifically, the part where `ValidatedTextInput` was moved to the checkout package) but keeps the part where styles, and a spread were added to `ValidatedTextInput` to handle the cases when this component is used as `type="number"`.

**⚠️ Please ignore the changes in `packages/checkout/utils/update-cart-from-api.ts`** - these are changed again in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4298 which is based off this branch.


Fixes #4205 




### How to test the changes in this Pull Request:

1. Check out `feature/add-inputs-for-points-redemption` from https://github.com/woocommerce/woocommerce-points-and-rewards/ and ensure you've run `npm run build` in that project.
2. Go to the file `includes/class-wc-points-rewards-extend-store-endpoint.php` in Points and Rewards and comment out the following lines (66-72):
```php
self::$extend->register_update_callback(
  array(
    'endpoint'          => CartExtensionsSchema::IDENTIFIER,
    'namespace'         => self::IDENTIFIER,
    'callback_function' => array( 'WC_Points_Rewards_Cart_Checkout', 'rest_apply_discount' ),
  )
);
```
3. Ensure you have added points to your user account ( WooCommerce -> Points and Rewards) and that you're logged in 
4. Add items to your cart and visit the Cart block. See that the Points and Rewards input is present. Note that it won't function on this branch. This PR is just to get the display working.



### Changelog

> Add Slot in the Discounts section of the cart sidebar to allow third party extensions to render their own components there.
